File: 05_Storage/02_Serializable/02_Parcelable/src/main/java/com/codingnight/example/parcelable/MainActivity.java
package com.codingnight.example.parcelable;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public class MainActivity extends AppCompatActivity {
    private static final String FILE_NAME = "test_file";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.button).setOnClickListener((v) -> {
            Student student = new Student("测试",10, new Score(90, 80, 70));
            Intent intent = new Intent(MainActivity.this, DetailActivity.class);
//            intent.putExtra("student", student);
            Bundle bundle = new Bundle();
            bundle.putParcelable("student", student);
            intent.putExtra("data", bundle);
            startActivity(intent);
        });
    }
}

File: 10_WorkManager/04_Workchain/src/main/java/com/codingnight/example/workmanager/MainActivity.kt
package com.codingnight.example.workmanager

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.work.OneTimeWorkRequest
import androidx.work.OneTimeWorkRequestBuilder
import androidx.work.WorkManager
import androidx.work.workDataOf

class MainActivity : AppCompatActivity() {
    private val workManager = WorkManager.getInstance(this)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View) {
        val workRequestA = createWork("Work A")
        val workRequestB = createWork("Work B")
        val workRequestC = createWork("Work C")
        workManager.beginWith(workRequestA)
            .then(workRequestB)
            .then(workRequestC)
            .enqueue()
    }

    private fun createWork(name: String): OneTimeWorkRequest {
        val workRequest = OneTimeWorkRequestBuilder<MyWorker>()
            .setInputData(workDataOf("IN_KEY" to name))
            .build()
        return workRequest
    }

    override fun onStop() {
        super.onStop()
        finish()
    }
}